ci/ci/build.py

```diff
@@ -422,19 +441,37 @@ def build(self, batch, code, scope):
 
         log.info(f'step {self.name}, rendered script:\n{rendered_script}')
 
+        copy_inputs = ''
+        input_files = None
         if self.inputs:
             input_files = []
             for i in self.inputs:
-                input_files.append((f'{BUCKET}/build/{batch.attributes["token"]}{i["from"]}', i["to"]))
-        else:
-            input_files = None
+                assert 'extract' not in i or i.get('directory') == 'archive'
```

Comment:
Why do you assert not extract here but support it below?

Reply:
same as above